source/dotnet/Test/AdaptiveCards.Test/AdaptiveCardApiTests.cs

```diff
@@ -1245,5 +1247,161 @@ public void TestParsingRichTextBlockWithInvalidInlineType()
+            var adaptiveCard =
+@"{
+	""$schema"": ""http://adaptivecards.io/schemas/adaptive-card.json"",
+	""type"": ""AdaptiveCard"",
+	""version"": ""1.0"",
+	""body"": [
+		{
+			""type"": ""TextBlock"",
+			""text"": ""Publish Adaptive Card schema"",
+			""weight"": ""bolder"",
+			""size"": ""medium""
+		},
+		{
+			""type"": ""ColumnSet"",
+			""columns"": [
+				{
+					""type"": ""Column"",
+					""width"": ""auto"",
+					""items"": [
+						{
+							""type"": ""Image"",
+							""url"": ""https://pbs.twimg.com/profile_images/3647943215/d7f12830b3c17a5a9e4afcc370e3a37e_400x400.jpeg"",
+							""size"": ""small"",
+							""style"": ""person""
+						}
+					]
+				},
+				{
+					""type"": ""Column"",
+					""width"": ""stretch"",
+					""items"": [
+						{
+							""type"": ""TextBlock"",
+							""text"": ""Matt Hidinger"",
+							""weight"": ""bolder"",
+							""wrap"": true
+						},
+						{
+							""type"": ""TextBlock"",
+							""spacing"": ""none"",
+							""text"": ""Created {{DATE(2017-02-14T06:08:39Z, SHORT)}}"",
+							""isSubtle"": true,
+							""wrap"": true
+						}
+					]
+				}
+			]
+		},
+		{
+			""type"": ""TextBlock"",
+			""text"": ""Now that we have defined the main rules and features of the format, we need to produce a schema and publish it to GitHub. The schema will be the starting point of our reference documentation."",
+			""wrap"": true
+		},
+		{
+			""type"": ""FactSet"",
+			""facts"": [
+				{
+					""title"": ""Board:"",
+					""value"": ""Adaptive Card""
+				},
+				{
+					""title"": ""List:"",
+					""value"": ""Backlog""
+				},
+				{
+					""title"": ""Assigned to:"",
+					""value"": ""Matt Hidinger""
+				},
+				{
+					""title"": ""Due date:"",
+					""value"": ""Not set""
+				}
+			]
+		}
+	],
+	""actions"": [
+		{
+			""type"": ""Action.ShowCard"",
+			""title"": ""Set due date"",
+			""card"": {
+				""type"": ""AdaptiveCard"",
+				""body"": [
+					{
+						""type"": ""Input.Date"",
+						""id"": ""dueDate""
+					}
+				],
+				""actions"": [
+					{
+						""type"": ""Action.Submit"",
+						""title"": ""OK""
+					}
+				]
+			}
+		},
+		{
+			""type"": ""Action.ShowCard"",
+			""title"": ""Comment"",
+			""card"": {
+				""type"": ""AdaptiveCard"",
+				""body"": [
+					{
+						""type"": ""Input.Text"",
+						""id"": ""comment"",
+						""isMultiline"": true,
+						""placeholder"": ""Enter your comment""
+					}
+				],
+				""actions"": [
+					{
+						""type"": ""Action.Submit"",
+						""title"": ""OK""
+					}
+				]
+			}
+		}
+	]
+}";
+            List<Thread> jobs = new List<Thread>();
+            for (var i = 0; i < 10; i++)
+            {
+                var payload = adaptiveCard;
+                Thread job = new Thread(() => RenderCardTask(payload));
+                jobs.Add(job);
+                job.Start();
+            }
+
+            foreach (var job in jobs)
+            {
+                job.Join();
```

Comment:
>job.Join(); [](start = 16, length = 11)

IF there is any unhandled exception anywhere in the workflow here there is a potential for thread.join to just block and hang the entire test process; Spinning up threads manually is also not typically needed now wiht the TPL.. Spin up tasks instead via the tasks parallel library (TPL) - look this up [https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/index](https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/index)

Reply:
exceptions are handled at LN1253-1263, I will look into TPL

---
In reply to: [340304111](https://github.com/microsoft/AdaptiveCards/pull/3546#discussion_r340304111) [](ancestors = 340304111)